activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/process/BpmnModelSequenceFlowValidator.java

```diff
@@ -52,14 +54,14 @@
         List<ModelValidationError> errors = new ArrayList<>();
         if (StringUtils.isEmpty(sequenceFlow.getSourceRef())) {
             errors.add(createModelValidationError(NO_SOURCE_REF_PROBLEM,
-                NO_SOURCE_REF_PROBLEM_DESCRIPTION,
-                SEQUENCE_FLOW_VALIDATOR_NAME));
+                format(NO_SOURCE_REF_PROBLEM_DESCRIPTION, sequenceFlow.getName(), sequenceFlow.getId()),
+                SEQUENCE_FLOW_VALIDATOR_NAME, sequenceFlow.getName()));
```

Comment:
I don't think you're calling the right method here. Is this supposed to set the new `referenceId`? here your setting the error code. Also, to set the reference id is better to use `sequenceFlow.getId()`.

Reply:
Sorry for the confusion from my end, most of the issues I fixed with the good commit.  Thanks.